### ISSUE TITLE
Testing farm??? 

### ISSUE THREAD

**Issue by johannbg** [packit/packit-service#1014]
Noticed that packit failed in our repo on rawhide and it seems like it due to somekind of testing farm feature being magically enabled for everyone.

It's failing/error-ing out on rawhide seemingly due to some ansible python path deprecation warning. 

http://artifacts.dev.testing-farm.io/4bd8b02a-63f1-444e-aeb2-8645419d874d/

Arguably you should not be blindly enable "features" in packit for everyone. 
Upstream/downstream maintainers should arguably be those that enable those if and then when they intend on using those features.

**Comment by johannbg** [comment]
Well the main problem here of course is that we are getting a failed test for something we did not, in a sense ask for. 

What this does is this leads to additional cost of paid/contributors time. The time it takes looking into this +  the cost of distraction ( CoD ) as in the PR submitter looks into why his or her PR fails, the reviewers looks into why the PR's for the submitter is failing ( and now you are looking into my issue due to this being failing, as well as @thrix looking into this and fix what is broken + the time of the reviewers of his fix and then rinse and repeat for more issues of similar nature like this ). 

On average a "ping" of some sort ( email, a phone call, coworker asking for something ) costs on average an half an hour, to an hour, factoring in the cost of distraction ( the point you got pinged, to the point you are back focusing on what you were doing prior to the ping ) and ofcourse meetings completely destroy any kind of productivity, which is why people should only schedule/attend meetings at the beginning ( 08:00 - 10:00 ) or end of the day ( 14:00 - 16:00 ) and never on Monday's and Friday's. 

Anyway it's best if people/projects can configure/enabled/disable ( within the dashboard or FAS ) what actually is run since in our case we currently only do our testing upstream and basically care for little to no other tests in Fedora other than "does it build" since if it does, it should just work.

**Comment by lachmanfrantisek** [comment]
> What this does is this leads to additional cost of paid/contributors time. 

Sure, but packit can't work alone. We mainly integrate other builders/test-runners/Fedora systems. Our aim is to hide this from you (=user) as much as possible but we can't avoid using them. But sure, we can provide better UX when something breaks on another side and we are improving this on the way.

> Anyway it's best if people/projects can configure/enabled/disable ( within the dashboard or FAS ) what actually is run since in our case we currently only do our testing upstream and basically care for little to no other tests in Fedora other than "does it build" since if it does, it should just work.

This is enabled by default but can be disabled by specifying jobs in your config:
* You can find more in [our documentation](https://packit.dev/docs/configuration/#packit-service-jobs).
* Doing this in the dashboard would be nice but probably not doable in the recent future.
* We agreed on turning this on by default after some discussion. It can bring some value without any work required. (Package builds does not mean package can be installed.)
* If it leads to a misleading page instead of providing pass/fail on checking installability, it's a bug on our side and we need to fix that. (Either by fixing this or by communicating with the testing-farm team. I am not sure if we can do better with this.)

**Comment by lachmanfrantisek** [comment]
> ...  in our case we currently only do our testing upstream and basically care for little to no other tests in Fedora other than "does it build" since if it does, it should just work.

Just to comment on this as well, this is not true for all. In Fedora (and on other distributions as well) you can get one version of your dependencies and can't do much about it. So for some projects (our Fedora packages including), it is really nice not only to try installing the package but also have some tests in that environment. It has saved us multiple times from pushing a non-compatible change to Fedora.

**Comment by johannbg** [comment]
Well the question is, brings value to whom? 

You do realize that you have to "sell" the idea of packit to upstream for it to be carried in the tree right so if the defaults are not tailored to upstream ( not downstream ) it can be a pretty hard thing to sell right. ( many upstreams dont even play distro's et all ) 

Now arguably the only value upstream sees in packit is if upstream maintainer(s) are also downstream maintainers and packit helps reduce the time he or she spends on downstream maintainership so that time can be better spent focusing on the project itself, or more projects if she or he is involved in more than one project ( which also is often the case) or to free time do some personal stuff. 

If packit increases existing contributors time ( as opposed to reduce it ) then it will simply end up being removed from tree's ( not implemented/used et al ). 

Anyway I suggest that you update the job section with what the "default" jobs that are run are ( also clearly document all defaults ) and how to disable/override them.

**Comment by lachmanfrantisek** [comment]
> You do realize that you have to "sell" the idea of packit to upstream for it to be carried in the tree right so if the defaults are not tailored to upstream ( not downstream ) it can be a pretty hard thing to sell right. ( many upstreams dont even play distro's et all )

Yes, selling distribution to upstream developers is hard in general. I hope I don't need to talk about the benefits of distributions, but if upstream cares about downstream, the ideal state is that their tooll/package/library works in an "expected way" in the distribution, what means:
- Package can be created/build. (=> copr build done by Packit)
- Package can be installed. (=> default test-case)
- Package works as expected in the distribution. (=> Upstream test-suite passes in the distribution environment. / If you run your tests in the testing-farm.)

I agree that UX can be improved, but don't think those checks are not relevant to "upstream-only" contributors.

> Anyway I suggest that you update the job section with what the "default" jobs that are run are ( also clearly document all defaults ) and how to disable/override them.

Sounds like a good suggestion for improvement.

-----

To conclude what can help in this situation:
- Users don't get links to different systems => everything goes through packit's page. (#506)
- Hide problems of the other systems we use internally. (This is ongoing work based on retries, checking of the pending states periodically,...)
- [ ] **Have clear documentation of the defaults. Make this information easily accessible to newcomers. (Task of this issue.)**
- Think about defaults (and all features) from the perspective of upstream-only contributors and what value we are giving and how do they understand it. This is very broad but important. (Not a task of this issue. It's a general rule.)

**Comment by lachmanfrantisek** [comment]
Can I change the title to match the actual task (=better docs for defaults) that needs to be done?

**Comment by thrix** [comment]
> This installability test is what failed. => It looks like some bug in the testing-farm itself, @thrix can you please take a look?

Yep, a workaround gone bad in case rawhide builds are removed from repository :( should be resolved from yesterday. I will change the workaround to work in a different way, so this does not happen anymore, sorry for the trouble.

**Comment by thrix** [comment]
Installation issue is hopefully gone forever.

**Comment by lachmanfrantisek** [comment]
@thrix thanks a lot for quick fix!

**Comment by lachmanfrantisek** [comment]
Looking at the current state of [docs](https://packit.dev/docs/configuration/#packit-service-jobs) so the info is there but what can be done to improve the current state:
* Default jobs can be easily found by newcomers. (Not sure how we can improve this.)
* Provide info about installability test (Maybe also in the dashboard on the result page to give some context right away.)
